src/core/actions/unitactions/Convert.java

```diff
@@ -22,19 +23,31 @@ public Unit getTarget() {
 
     @Override
     public LinkedList<Action> computeActionVariants(final GameState gs) {
-        //TODO: compute all the Convert actions that are possible
-        return null;
+        LinkedList<Action> converts = new LinkedList<>();
+        Board b = gs.getBoard();
+        if(isFeasible(gs)){
+            Convert a = new Convert(this.unit, this.target);
+            converts.add(a);
+        }
+        return converts;
     }
 
     @Override
     public boolean isFeasible(final GameState gs) {
-        //TODO: check if this Convert action is possible.
+        //Check if target in range
+        if(target.getCurrentPosition().x < unit.getCurrentPosition().x  + this.unit.RANGE && target.getCurrentPosition().y < unit.getCurrentPosition().y  + this.unit.RANGE)
+            return true;
         return false;
     }
 
     @Override
     public boolean execute(GameState gs) {
-        //TODO execute the Convert action
+        //Check if action is feasible before execution
+        if(isFeasible(gs)) {
+            target.setActorID(unit.getActorID());
```

Comment:
ActorID is unique per entity in the game. Once they're set, they never change. This line should be removed.